### PR TITLE
core: convert gather-runner.js to async/await

### PR DESCRIPTION
**Pull request by JoelEinbinder**
I was reading through how this file worked, and decided to convert it to async/await as I did. Clears up some dastardly uses of `array.reduce`.

### CODE REVIEW

File: lighthouse-core/gather/gather-runner.js [hunk 1]
```diff
@@ -248,31 +242,25 @@ class GatherRunner {
     const status = 'Loading page & waiting for onload';
     log.log('status', status, gatherernames);
 
-    const pass = Promise.resolve()
-      // Clear disk & memory cache if it's a perf run
-      .then(_ => {
-        if (isPerfRun) driver.cleanBrowserCaches();
```
Comment (JoelEinbinder) [inline lighthouse-core/gather/gather-runner.js]: this code wasn't be awaited, but it looks like it should be. Same with `beginTrace` down below.

File: lighthouse-core/gather/gather-runner.js [hunk 2]
```diff
@@ -248,31 +242,25 @@ class GatherRunner {
...
+    // Clear disk & memory cache if it's a perf run
+    if (isPerfRun) await driver.cleanBrowserCaches();
+    // Always record devtoolsLog
+    await driver.beginDevtoolsLog();
+    // Begin tracing if requested by config.
+    if (recordTrace) await driver.beginTrace(settings);
+
+    // Navigate.
+    await GatherRunner.loadPage(driver, passContext);
+    await log.log('statusEnd', status);
```
Comment (patrickhulce) [inline lighthouse-core/gather/gather-runner.js]: this one shouldn't need to be awaited

File: lighthouse-core/gather/gather-runner.js (hunk 1 above)
Comment (brendankenny) [reply lighthouse-core/gather/gather-runner.js]: maybe https://palantir.github.io/tslint/rules/no-floating-promises/, which isn't perfect but would have caught this

Comment (JoelEinbinder) [reply lighthouse-core/gather/gather-runner.js]: I think just using async/await makes these stand out a lot more. That lint rule might be more trouble than its worth.

File: lighthouse-core/gather/gather-runner.js (hunk 2 above)
Comment (JoelEinbinder) [reply lighthouse-core/gather/gather-runner.js]: done.

**Pull request merged**